examples/tldraw-example/src/ui-options.tsx

```diff
@@ -1,16 +1,71 @@
+import { TDShapeType, TDToolType, Tldraw, TldrawApp } from '@tldraw/tldraw'
 import * as React from 'react'
-import { Tldraw } from '@tldraw/tldraw'
+
+function SelectButton({
+  type,
+  tldrawApp,
+  children,
+}: React.PropsWithChildren<{
+  type: TDToolType
+  tldrawApp?: TldrawApp
+}>) {
+  return (
+    <button
+      onClick={() => {
+        tldrawApp?.selectTool(type)
+      }}
+      style={{
+        border: '1px solid #333',
+        // @TODO: have the button re-render when tldrawApp.currentTool.type changes, else active states won't work
+        background: tldrawApp?.currentTool?.type === type ? 'papayawhip' : 'transparent',
```

Comment:
there is a state activeTool in tldrawApp you can use! pass it as a prop
more info [activeTool](https://github.com/tldraw/tldraw/blob/main/packages/tldraw/src/components/ToolsPanel/PrimaryTools.tsx#L17)